inc/targetproblem.class.php

```diff
@@ -0,0 +1,727 @@
+      $domain = PluginFormcreatorForm::getTranslationDomain($form->getID());
+      $data['name'] = $this->prepareTemplate(
+         __($this->fields['target_name'], $domain),
+         $formanswer,
+         true
+      );
+      $data['name'] = Toolbox::addslashes_deep($data['name']);
+      $data['name'] = $formanswer->parseTags($data['name'], $this);
+
+      $problemFields = [
+         'content',
+         'impactcontent',
+         'causecontent',
+         'symptomcontent',
+      ];
+      foreach ($problemFields as $problemFields) {
+         $data[$problemFields] = $this->prepareTemplate(
+            $this->fields[$problemFields],
+            $formanswer,
+            $problemFields == 'content' // only content supports rich text
+         );
+         $data[$problemFields] = $data[$problemFields] ?? '';
+
+         $data[$problemFields] = $formanswer->parseTags($data[$problemFields], $this, $problemFields == 'content');
+      }
+
+      $data['_users_id_recipient']   = $_SESSION['glpiID'];
+
+      $this->prepareActors($form, $formanswer);
+
+      if (count($this->requesters['_users_id_requester']) == 0) {
+         $this->addActor('requester', $formanswer->fields['requester_id'], true);
+         $requesters_id = $formanswer->fields['requester_id'];
+      } else {
+         $requesterAccounts = array_filter($this->requesters['_users_id_requester'], function($v) {
+            return ($v != 0);
+         });
+         $requesters_id = array_shift($requesterAccounts);
+         if ($requesters_id === null) {
+            // No account for requesters, then fallback on the account used to fill the answers
+            $requesters_id = $formanswer->fields['requester_id'];
+         }
+      }
+
+      $data = $this->setTargetEntity($data, $formanswer, $requesters_id);
+      $data = $this->setTargetUrgency($data, $formanswer);
+
+      $data = $this->requesters + $this->observers + $this->assigned + $this->assignedSuppliers + $data;
+      $data = $this->requesterGroups + $this->observerGroups + $this->assignedGroups + $data;
+
+      // Create the target problem
+      if (!$problemID = $problem->add($data)) {
+         return null;
+      }
+
+      $this->saveTags($formanswer, $problemID);
+
+      // Add link between Problem and FormAnswer
+      $itemlink = $this->getItem_Item();
+      $itemlink->add([
+         'itemtype'     => PluginFormcreatorFormAnswer::class,
+         'items_id'     => $formanswer->fields['id'],
+         'problems_id'  => $problemID,
+      ]);
+
+      $this->attachDocument($formanswer->getID(), Problem::class, $problem);
+
+      return $problem;
+   }
+
+   protected function getTaggableFields() {
+      return [
+         'target_name',
+         'content',
+         'impactcontent',
+         'causecontent',
+         'symptomcontent',
+      ];
+   }
+
+   public function prepareInputForUpdate($input) {
+      // Control fields values :
+      if (!isset($input['_skip_checks'])
```

Comment:
Is it safe to handle `_skip_checks` directly in the input ?
Wouldn't it be possible for a malicious user to insert it into his form and save unchecked data ?